demisto_sdk/__main__.py

```diff
@@ -1956,40 +1956,52 @@ def init(ctx, **kwargs):
 )
+@click.option(
+    "-rt",
+    "--readme-template",
+    help="The readme template that should be appended to the given README.md file",
+    type=click.Choice(["syslog", "xdrc", "http-collector"]),
+)
 @click.pass_context
 @logging_setup_decorator
 def generate_docs(ctx, **kwargs):
     """Generate documentation for integration, playbook or script from yaml file."""
-    check_configuration_file("generate-docs", kwargs)
-    input_path_str: str = kwargs.get("input", "")
-    if not (input_path := Path(input_path_str)).exists():
-        logger.info(f"[red]input {input_path_str} does not exist[/red]")
-        return 1
+    try:
+        check_configuration_file("generate-docs", kwargs)
+        input_path_str: str = kwargs.get("input", "")
+        if not (input_path := Path(input_path_str)).exists():
+            raise Exception(f"[red]input {input_path_str} does not exist[/red]")
+
+        if (output_path := kwargs.get("output")) and not Path(output_path).is_dir():
+            raise Exception(
+                f"[red]Output directory {output_path} is not a directory.[/red]"
+            )
 
-    if (output_path := kwargs.get("output")) and not Path(output_path).is_dir():
-        logger.info(f"[red]Output directory {output_path} is not a directory.[/red]")
-        return 1
+        if input_path.is_file():
+            if input_path.suffix.lower() not in {".yml", ".md"}:
+                raise Exception(
+                    f"[red]input {input_path} is not a valid yml or readme file.[/red]"
+                )
 
-    if input_path.is_file():
-        if input_path.suffix.lower() != ".yml":
-            logger.info(f"[red]input {input_path} is not a valid yml file.[/red]")
-            return 1
-        _generate_docs_for_file(kwargs)
+            _generate_docs_for_file(kwargs)
 
-    # Add support for input which is a Playbooks directory and not a single yml file
-    elif input_path.is_dir() and input_path.name == "Playbooks":
-        for yml in input_path.glob("*.yml"):
-            file_kwargs = copy.deepcopy(kwargs)
-            file_kwargs["input"] = str(yml)
-            _generate_docs_for_file(file_kwargs)
+        # Add support for input which is a Playbooks directory and not a single yml file
+        elif input_path.is_dir() and input_path.name == "Playbooks":
+            for yml in input_path.glob("*.yml"):
+                file_kwargs = copy.deepcopy(kwargs)
+                file_kwargs["input"] = str(yml)
+                _generate_docs_for_file(file_kwargs)
 
-    else:
-        logger.info(
-            f"[red]Input {input_path} is neither a valid yml file, nor a folder named Playbooks."
-        )
-        return 1
+        else:
+            raise Exception(
+                f"[red]Input {input_path} is neither a valid yml file, nor a folder named Playbooks, nor a readme file."
+            )
 
-    return 0
+        return 0
+
+    except Exception as err:
+        logger.error(err)
```

Comment:
```suggestion
        logger.exception("failed generating docs")
```
This [adds context](https://www.loggly.com/blog/exceptional-logging-of-exceptions-in-python/) automatically